dockstore-common/src/main/java/io/dockstore/common/NextflowUtilities.java

```diff
@@ -0,0 +1,150 @@
+import org.apache.commons.configuration2.ConfigurationConverter;
+import org.apache.commons.configuration2.INIConfiguration;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class NextflowUtilities {
+    private static final Logger LOG = LoggerFactory.getLogger(NextflowUtilities.class);
+    private static final String DEFAULT_NEXTFLOW_VERSION = "19.01.0";
+
+    private NextflowUtilities() {
+        // hide the default constructor for a utility class
+    }
+
+    public static File getNextFlowTargetFile(INIConfiguration config) {
+        String nextflowVersion = config.getString("nextflow-version", DEFAULT_NEXTFLOW_VERSION);
+        return getNextFlowTargetFile(nextflowVersion);
+    }
+
+    private static File getNextFlowTargetFile() {
+        return getNextFlowTargetFile(DEFAULT_NEXTFLOW_VERSION);
+    }
+
+    private static File getNextFlowTargetFile(String nextflowVersion) {
+        String nextflowExec =
+            "https://github.com/nextflow-io/nextflow/releases/download/v" + nextflowVersion + "/nextflow-" + nextflowVersion + "-all";
+        if (!Objects.equals(DEFAULT_NEXTFLOW_VERSION, nextflowVersion)) {
+            System.out.println("Running with Nextflow " + nextflowVersion + " , Dockstore tests with " + DEFAULT_NEXTFLOW_VERSION);
+        }
+
+        // grab the nextflow jar if needed
+        String libraryLocation =
+            System.getProperty("user.home") + File.separator + ".dockstore" + File.separator + "libraries" + File.separator;
+        URL nextflowURL;
+        String nextflowFilename;
+        try {
+            nextflowURL = new URL(nextflowExec);
+            nextflowFilename = new File(nextflowURL.toURI().getPath()).getName();
+        } catch (MalformedURLException | URISyntaxException e) {
+            throw new NextflowParsingException("Could not create Nextflow location", e);
+        }
+        String nextflowTarget = libraryLocation + nextflowFilename;
+        File nextflowTargetFile = new File(nextflowTarget);
+        if (!nextflowTargetFile.exists()) {
+            try {
+                FileUtils.copyURLToFile(nextflowURL, nextflowTargetFile);
+            } catch (IOException e) {
+                throw new NextflowParsingException("Could not download Nextflow location", e);
+            }
+        }
+        return nextflowTargetFile;
+    }
+
+    /**
+     * Use nextflow to read nextflow configs
+     * Relies on content being in it's own directory
+     * TODO: nextflow normally uses this, investigate to see if this helps us https://github.com/ninjudd/drip
+     * @param content a file object with the content of nextflow
+     * @return a commons configuration file with the keys from the nextflow config file
+     */
+    public static Configuration grabConfig(File content) {
+        try {
+            final ArrayList<String> strings = new ArrayList<>(
```

Comment:
You can just use Arrays.asList directly, e.g., `final List<String> strings = Arrays.asList(...`